.github/workflows/test-and-deploy.yaml

```diff
@@ -19,6 +19,9 @@ jobs:
         uses: actions/checkout@v3
       - name: Create Docker network
         run: docker network create uccser-development-stack
+        # Required for the node service
+      - name: Set DOCKER_UID variable
+        run: echo "DOCKER_UID=$(echo $UID)" >> $GITHUB_ENV
```

Comment:
What is the value appended to `$GITHUB_ENV`?

Reply:
That line is essentially the same as this line in `./dev`: https://github.com/uccser/cs-field-guide/pull/2011/files#diff-dc2babd290214832274e2012c4d8af5cb43bcd1e5f3a8aa7955d6143d37b407aR4. It is needed because some jobs call `docker compose up` manually, rather than through `./dev`

Reply:
Also, the reason we need to set the variable at all, is because $UID is a bash variable and is not exposed to docker, so I need to wrap it in a environment variable

Reply:
@JackMorganNZ

Reply:
Sorry, my question is why is it not just: `run: echo "DOCKER_UID=$(echo $UID)"`.

The appending at the end doesn't seem to add anything in our pipeline.

Reply:
Oh, that is just how you set environment variables inside a github actions job.

Reply:
But why does the environment variable need to be set?

Reply:
Because this line now depends on the current users UID, this line will fail if the environment variable is not set. I have no idea why the default parameter in the line above doesn't work, but it doesn't, so we need the env variable.

Reply:
I found the doc page that explained it: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

This allows it to be used by subsequent steps in the job.